model/jpa/src/main/java/org/keycloak/models/jpa/session/JpaUserSessionPersisterProvider.java

```diff
@@ -228,6 +228,11 @@ public void updateLastSessionRefreshes(RealmModel realm, int lastSessionRefresh,
     public void removeExpired(RealmModel realm) {
         int expiredOffline = Time.currentTime() - realm.getOfflineSessionIdleTimeout() - SessionTimeoutHelper.PERIODIC_CLEANER_IDLE_TIMEOUT_WINDOW_SECONDS;
 
+        // prefer client session timeout if set
+        if (realm.getClientOfflineSessionIdleTimeout() > 0) {
+            expiredOffline = Time.currentTime() - realm.getClientOfflineSessionIdleTimeout() - SessionTimeoutHelper.PERIODIC_CLEANER_IDLE_TIMEOUT_WINDOW_SECONDS;
```

Comment:
Should this timeout apply only to client sessions?

Reply:
I don't think it will work:

1. getClientOfflineSessionIdleTimeout not set == 0 -> it will take second argument as current time - 0 will be always greater but we want first argument
2. getClientOfflineSessionIdleTimeout is set and it's greater than getOfflineSessionIdleTimeout -> it will take first argument but again we want second

Reply:
+1, it won't, the corresponding timeout should apply to respective user / client sessions